pkg/domain/infra/abi/images.go

```diff
@@ -748,28 +749,38 @@ func (ir *ImageEngine) Sign(ctx context.Context, names []string, options entitie
 				return err
 			}
 
-			// create signature
-			newSig, err := signature.SignDockerManifest(getManifest, dockerReference.String(), mech, options.SignBy)
-			if err != nil {
-				return errors.Wrapf(err, "error creating new signature")
-			}
-			// create the signstore file
-			signatureDir := fmt.Sprintf("%s@%s=%s", sigStoreDir, manifestDigest.Algorithm(), manifestDigest.Hex())
-			if err := os.MkdirAll(signatureDir, 0751); err != nil {
-				// The directory is allowed to exist
-				if !os.IsExist(err) {
-					logrus.Error(err)
-					return nil
+			if options.All {
+				list, err := manifest.ListFromBlob(getManifest, manifestType)
```

Comment:
Probably check `manifest.MIMETypeIsMultiImage` first to give the user a meaningful error.